libbeat/statestore/backend/memlog/memlog_test.go

```diff
@@ -0,0 +1,258 @@
+// Licensed to Elasticsearch B.V. under one or more contributor
+// license agreements. See the NOTICE file distributed with
+// this work for additional information regarding copyright
+// ownership. Elasticsearch B.V. licenses this file to you under
+// the Apache License, Version 2.0 (the "License"); you may
+// not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package memlog
+
+import (
+	"encoding/json"
+	"io"
+	"io/ioutil"
+	"math"
+	"os"
+	"path/filepath"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/beats/v7/libbeat/logp"
+	"github.com/elastic/beats/v7/libbeat/statestore"
+	"github.com/elastic/beats/v7/libbeat/statestore/backend"
+	"github.com/elastic/beats/v7/libbeat/statestore/internal/storecompliance"
+)
+
+func init() {
+	logp.DevelopmentSetup()
+}
+
+func TestCompliance_Default(t *testing.T) {
+	storecompliance.TestBackendCompliance(t, func(testPath string) (backend.Registry, error) {
+		return New(logp.NewLogger("test"), Settings{Root: testPath})
+	})
+}
+
+func TestCompliance_AlwaysCheckpoint(t *testing.T) {
+	storecompliance.TestBackendCompliance(t, func(testPath string) (backend.Registry, error) {
+		return New(logp.NewLogger("test"), Settings{
+			Root: testPath,
+			Checkpoint: func(filesize uint64) bool {
+				return true
+			},
+		})
+	})
+}
+
+func TestLoadVersion1(t *testing.T) {
+	dataHome := "testdata/1"
+
+	list, err := ioutil.ReadDir(dataHome)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	cases := list[:0]
```

Comment:
I'm not sure I understand this construction -- this should be equivalent to `cases := []os.FileInfo{}`, right? Is this a type inference shortcut in the spirit of c++ `auto`, or something else?

Reply:
>  this should be equivalent to cases := []os.FileInfo{}, right?

kinda. As we have slices the construct creates a slice of length 0, but with the old capacity.
It's an in-place filtering loop for removing files from the list (e.g. the Readme.md). See: https://github.com/golang/go/wiki/SliceTricks#filtering-without-allocating

But yeah, this is me being lazy :)